packages/medusa/src/api/routes/store/carts/create-line-item.js

```diff
@@ -43,27 +43,27 @@ export default async (req, res) => {
     const lineItemService = req.scope.resolve("lineItemService")
     const cartService = req.scope.resolve("cartService")
 
-    let cart = await cartService.retrieve(id)
-
     await manager.transaction(async m => {
-      const line = await lineItemService.generate(
-        value.variant_id,
-        cart.region_id,
-        value.quantity,
-        { metadata: value.metadata }
-      )
-      await cartService.withTransaction(m).addLineItem(id, line)
+      const txCartService = cartService.withTransaction(m)
+      const cart = await txCartService.retrieve(id)
+
+      const line = await lineItemService
+        .withTransaction(m)
+        .generate(value.variant_id, cart.region_id, value.quantity, {
+          metadata: value.metadata,
+        })
+      await txCartService.addLineItem(id, line)
 
-      const updated = await cartService.withTransaction(m).retrieve(id, {
+      const updated = await txCartService.retrieve(id, {
         relations: ["payment_sessions"],
       })
 
       if (updated.payment_sessions?.length) {
-        await cartService.withTransaction(m).setPaymentSessions(id)
+        await txCartService.setPaymentSessions(id)
       }
     })
 
-    cart = await cartService.retrieve(id, {
+    const cart = await cartService.retrieve(id, {
```

Comment:
Is this meant to be part of this PR?

Reply:
SQLite only allows one transaction at a time pr connection, which is also what we want to have within the scope of each endpoint. I found out that this endpoint used two transactions as the `lineItemService` didn't use the transaction started in the controller. 

This fix should also eliminate the "QueryRunnerAlreadyReleased" errors that we sometimes experience on this endpoint with Postgres.

Reply:
This is great!